### PR TITLE
Compare baselines xpvwrh

### PR DESCRIPTION
**Pull request by Dariyanikitin**
tweaked extract flows and compare_baselines to work for the new analyis

### CODE REVIEW

File: R/run_grid.R [hunk 1]
```diff
@@ -69,6 +69,191 @@ run_grid  <- function(gono_params, init_params, cost_params,
...
+  # all vbe get two doses
+  ret$inc_vbe_doses <- ret$inc_vbe * 2
+
+  # calculate doses given per person offered primary vaccination
+  ret$inc_primary_full_doses <- ret$inc_primary * uptake_first_dose *
+                                                        uptake_second_dose
+  ret$inc_primary_part_doses <- ret$inc_primary * uptake_first_dose *
+                                                        (1 - uptake_second_dose)
+  ret$inc_primary_total_doses <- ret$inc_primary_full_doses +
+                                            ret$inc_primary_part_doses
```
Comment (lwhittles) [inline R/run_grid.R:211]:
```suggestion
  ret$inc_primary_full_doses <- ret$inc_primary *  uptake_second_dose * 2
  ret$inc_primary_part_doses <- ret$inc_primary * (1 - uptake_second_dose)
  ret$inc_primary_total_doses <- ret$inc_primary_full_doses + ret$inc_primary_part_doses
```
first dose uptake should already be applied here, i.e. the number of peope moving from X to either P or V is number offered * first dose uptake. Then it's second dose uptake that determines whether they get 1 dose or 2.

Also, I don't think you multiplied the full doses by 2

File: R/support.R [hunk 2]
```diff
@@ -20,6 +20,50 @@ aggregate <- function(x, what, as_incid = FALSE, stratum = NULL,
...
+  cumulative_flows$cum_part_to_full <-
+    t(aggregate(y, "cum_vaccinated", stratum = 2))
+
+  # revaccinated = everyone vaccinated from W(4), note does not include
+  # vaccination of individuals in X who have waned from P
+  # in earlier models this will be strata (3) but by subtracting primary
+  # and partial we don't need to specify the index
+  cumulative_flows$cum_revaccinated <-
+    cumulative_flows$cum_vaccinated - cumulative_flows$cum_primary_total
+                            - cumulative_flows$cum_part_to_full
```
Comment (lwhittles) [inline R/support.R:54]: Have you checked anywhere that this is the case? i.e. that cum_revaccinated = t(aggregate(y, "cum_vaccinated", stratum = 4))?

File: tests/testthat/test-run_grid.R [hunk 3]
```diff
@@ -14,6 +14,193 @@ test_that("calc_pv works as expected", {
...
+  expect_equal(z[flownames], yy)
+  expect_equivalent(z[incnames],
+                    Map(`-`, yy[flownames], bl[flownames]))
+
+  f <- function(x) {
+    pv <- x$vaccinated - x$revaccinated - x$vbe - x$part_to_full
+    (pv * r1 * r2) + (pv * r1 * (1 - r2)) + 1 * x$revaccinated +
+                                            1 * x$part_to_full +
+                                            2 * x$vbe
+  }
```
Comment (lwhittles) [inline tests/testthat/test-run_grid.R:56]: I think this check is incorrect as well (for reasons outlined above) To make sure this doesn't happen again, can you put in a check where second dose uptake = 100%, no revaccination and no P -> V, then the total doses should be 2 x number of vaccinations.

Similarly, can you add a check where 2nd dose uptake = 0%, no VBE, then total doses should = total vaccinations.

**Review by lwhittles — CHANGES_REQUESTED** [review]

File: R/support.R (hunk 2 above)
Comment (Dariyanikitin) [reply R/support.R:54]: I hadn't but i think its simpler just to change what i've done to   what you're suggesting as its less convoluted :)

**Review by lwhittles — APPROVED** [review]
looks good to me!

**Pull request merged**